ddtrace/contrib/internal/avro/schema_iterator.py

```diff
@@ -0,0 +1,141 @@
+import avro
+from avro.schema import Schema as AvroSchema
+
+from ddtrace._trace.span import Span
+from ddtrace.ext import schema as SCHEMA_TAGS
+from ddtrace.internal.datastreams import data_streams_processor
+from ddtrace.internal.datastreams.schemas.schema_builder import SchemaBuilder
+from ddtrace.internal.datastreams.schemas.schema_iterator import SchemaIterator
+
+
+class SchemaExtractor(SchemaIterator):
+    SERIALIZATION = "serialization"
+    DESERIALIZATION = "deserialization"
+    AVRO = "avro"
+
+    def __init__(self, schema: AvroSchema):
+        self.schema = schema
+
+    @staticmethod
+    def extract_property(field: avro, schema_name: str, field_name: str, builder: SchemaBuilder, depth: int) -> int:
+        array = False
+        type_ = None
+        format_ = None
+        description = None
+        ref = None
+        enum_values = None
+
+        field_type = field.type
+
+        if isinstance(field_type, list):
+            # Union type
+            array = True
+            type_ = SchemaExtractor.get_type(field_type[0].type)
+            for sub_type in field_type:
+                if isinstance(sub_type, dict) and sub_type.get("type") == "array":
+                    type_ = "array"
+                    break
+        elif field_type.type == "union":
```

Comment:
If the `field_type` was a primitive, then wouldn't this throw an `AttributeError`? Should there be an extra check that it has a `type` attribute first?

Reply:
I changed this to be type, though `field_type` is generally an Avro `FieldType` class object, so it always should have that `type` attribute.